File: leetcode/easy/find-smallest-letter-greater-than-target.go
package main

import "fmt"

func nextGreatestLetter(letters []byte, target byte) byte {
	for _, c := range letters {
		if target < c {
			return c
		}
	}

	return letters[0]
}

func main() {
	fmt.Println(nextGreatestLetter([]byte{'c', 'f', 'j'}, 'a'))
	fmt.Println(nextGreatestLetter([]byte{'c', 'f', 'j'}, 'c'))
	fmt.Println(nextGreatestLetter([]byte{'x', 'x', 'y', 'y'}, 'z'))
}

File: leetcode/easy/find-smallest-letter-greater-than-target.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

char nextGreatestLetter(vector<char>& letters, char target) {
    for (int i = 0; i < letters.size(); i++) {
        if (target < letters[i]) {
            return letters[i];
        }
    }

    return letters[0];
}


int main(void) {
	vector<char> arr{'c', 'f', 'j'};

	cout << nextGreatestLetter(arr, 'a') << endl;
}